test/files/pos/t5183.scala

```diff
@@ -0,0 +1,33 @@
+trait Day
+
+object Test {
+  def foo(t: Int with Day) = t == t
+}
+
+class DayOps(val i: Int with Day) extends AnyVal
+
+case class Test1(d: Int with Day)
+case class Test2(d1: Int with Day, d2: Int with Day)
+
+
+class User
+class Checkin
+  
+type Tagged[U] = { type Tag = U }
```

Comment:
scala doesn't allow top-level type aliases; this needs to be nested in some object